tests/language-feature/tuple/tuple-basic.slang

```diff
@@ -0,0 +1,25 @@
+// Test tuple swizzling and element access.
+
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-shaderobj
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-dx12 -use-dxil -shaderobj
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-cpu -shaderobj
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-vk -shaderobj
+
+//TEST_INPUT:ubuffer(data=[1 2 3 4], stride=4):out,name=outputBuffer
+RWStructuredBuffer<int> outputBuffer;
+
+void store(inout Tuple<int, float, uint> t)
+{
+    t._1_2 = Tuple<float, uint>(3.0, 4);
+}
+
+[numthreads(4, 1, 1)]
+void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
+{
+    var t = Tuple<int, float, uint>(1, 2.0, 3u);
```

Comment:
Unrelated to this code but this made me to wonder if we need to worry about alignment when the types are, as an example, like following,
```
Tuple<int8_t, float> // float values may not be aligned???
```

Reply:
A tuple type is equivalent to a struct type with the same typed fields. So it is equivalent to `struct S { int8_t a; float b};` and will follow struct layout rules, which means under natural layout `b` will be at offset 4.